celery/result.py

```diff
@@ -215,7 +215,7 @@ def get(self, timeout=None, propagate=True, interval=0.5,
         if on_interval:
             _on_interval.then(on_interval)
 
-        if self._cache:
+        if self._cache and not no_ack:
```

Comment:
can you please add a unit test to verify this change?

Reply:
I have no idea about unit test, can you explain how to do this?

Reply:
To explain a little more in detail, if it becomes a cache through the ready function, no_ack is False and the consumer operates. After that, through the get function, the consumer should operate with no_ack set to True, but the cache value is returned and the function is terminated without operating the consumer with no_ack set to True.

I don't know how make exactly test code.  I'm not sure how to write code that correct tests this phenomenon.